src/main/java/com/google/api/codegen/discovery/transformer/java/JavaSampleMethodToViewTransformer.java

```diff
@@ -85,19 +85,29 @@ private SampleView createSampleView(SampleTransformerContext context) {
         typeTable.getRequestTypeName(config.apiTypeName(), methodInfo.requestType()).getNickname();
 
     List<SampleFieldView> fields = new ArrayList<>();
+    List<SampleFieldView> optionalFields = new ArrayList<>();
     List<String> fieldVarNames = new ArrayList<>();
     for (FieldInfo field : methodInfo.fields().values()) {
-      SampleFieldView sampleFieldView = createSampleFieldView(field, typeTable, symbolTable);
+      SampleFieldView sampleFieldView = createSampleFieldView(field, context, symbolTable);
       fields.add(sampleFieldView);
-      fieldVarNames.add(sampleFieldView.name());
+      if (sampleFieldView.required()) {
+        fieldVarNames.add(sampleFieldView.name());
+      } else {
+        optionalFields.add(sampleFieldView);
+      }
```

Comment:
I wonder if it would make sense to call these "parameters" like the API docs, instead of "fields" like in protobuf, which has always seemed confusing to me ... for instance,

* `fields` -> `requiredParams`
* `fieldVarNames` -> `methodCallParams`
* `optionalFields` -> `optionalParams`.

But if that's too inconsistent with other usage, I think your suggestion is fine.